cmake_format/formatter.py

```diff
@@ -68,24 +68,48 @@ def format_comment_lines(node, config, line_width):
   for token in node.children:
     assert isinstance(token, lexer.Token)
     if token.type == TokenType.COMMENT:
-      inlines.append(token.spelling.strip().lstrip('#'))
+      # If this is a long enough sequence of ####, it gets set to line_width
+      if config.keep_comment_line_blocks and \
+          config.line_block_regex.match(token.spelling):
+        # NOTE: see also _refill_comment_line_blocks, that is where these two
+        # get added back.  Removed here because they will get reflowed.
+        inlines.append('#' * (line_width - 2))
+      else:
+        inlines.append(token.spelling.strip().lstrip('#'))
+
+  def early():
```

Comment:
This is a lot of specialized logic which replicates the [existing logic][1] for parsing/formatting rulers. I think it can all be replaced by simply specifying an additional `ruler_pattern`

[1]: https://cmake-format.readthedocs.io/en/latest/#markup